packages/core/src/boot/init.ts

```diff
@@ -48,6 +49,9 @@ export interface BuildEnv {
 }
 
 export function commonInit(userConfiguration: UserConfiguration, buildEnv: BuildEnv) {
+  if (userConfiguration.beforeSend) {
+    userConfiguration.beforeSend = catchErrors(userConfiguration.beforeSend, 'beforeSend threw an error:')
+  }
```

Comment:
IMO, this change is not at the right level of abstraction here, I'd move it inside the buildConfiguration function